### PR TITLE
Add filtering docs to Logger

### PR DESCRIPTION
**Pull request by atomkirk**

### CODE REVIEW

File: lib/logger/lib/logger.ex [hunk 1]
```diff
@@ -366,6 +366,31 @@ defmodule Logger do
   It sets up a log handler with overload protection and allows incoming events
   to be dispatched to multiple backends.
 
+  ### Filtering
+
+  You can add filters to Erlang's `:logger`. For example, to filter out logs
+  that contain a string, you could create a module:
```
Comment (whatyouhide) [inline lib/logger/lib/logger.ex:372]:
```suggestion
  that contain a particular string, you could create a module:
```

File: lib/logger/lib/logger.ex [hunk 2]
```diff
@@ -366,6 +366,31 @@ defmodule Logger do
...
+        @spec filter(map(), keyword()) :: :stop | :ignore | map()
+        def filter(log_event, _opts) do
+          case log_event do
+            %{msg: msg} when is_binary(msg) ->
+              if msg =~ "twitter" do
+                :stop
+              else
+                :ignore
+              end
+            _ ->
```
Comment (whatyouhide) [inline lib/logger/lib/logger.ex:384]:
```suggestion

            _ ->
```

File: lib/logger/lib/logger.ex [hunk 3]
```diff
@@ -366,6 +366,31 @@ defmodule Logger do
...
+              :ignore
+          end
+        end
+      end
+
+  Then, in your Application.start/2 function in application.ex:
```
Comment (whatyouhide) [inline lib/logger/lib/logger.ex:390]:
```suggestion
  Then, when you start your application, such as in the
  `c:Application.start/2` callback:
```

**Review by whatyouhide — APPROVED** [review]

File: lib/logger/lib/logger.ex [hunk 4]
```diff
@@ -366,6 +366,33 @@ defmodule Logger do
...
+
+  You can add filters to Erlang's `:logger`. For example, to filter out logs
+  that contain a particular string, you could create a module:
+
+      defmodule LogFilter do
+        @spec filter(map(), keyword()) :: :stop | :ignore | map()
+        def filter(log_event, _opts) do
+          case log_event do
+            %{msg: msg} when is_binary(msg) ->
+              if msg =~ "twitter" do
```
Comment (josevalim) [inline lib/logger/lib/logger.ex:379]:
```suggestion
              if msg =~ "password" do
```

File: lib/logger/lib/logger.ex [hunk 5]
```diff
@@ -366,6 +366,33 @@ defmodule Logger do
   It sets up a log handler with overload protection and allows incoming events
   to be dispatched to multiple backends.
 
+  ### Filtering
+
+  You can add filters to Erlang's `:logger`. For example, to filter out logs
+  that contain a particular string, you could create a module:
+
+      defmodule LogFilter do
+        @spec filter(map(), keyword()) :: :stop | :ignore | map()
```
Comment (josevalim) [inline lib/logger/lib/logger.ex:375]:
```suggestion
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

File: lib/logger/lib/logger.ex (hunk 4 above)
Comment (atomkirk) [reply lib/logger/lib/logger.ex]: oh come on

**Pull request merged**